### DISCUSSION TITLE
Layouts com float ou position?

### DISCUSSION THREAD

**Discussion by vitocaio** [frontendbr/forum#219] [Perguntas e Respostas]
Ola pessoal! tenho uma duvida, é melhor criar layouts utilizando float ou position? sei que existe diferenças entre os dois e que tudo depende do contexto, mas no geral um é usado de forma mais coerente que o outro, qual seria melhor e por que?

**Comment by kazzkiq** [comment]
**Nem um, nem outro.**

Se você não precisa dar suporte para navegadores mais antigos do que o IE 9, eu recomendaria pular essas alternativas e utilizar **FlexBox**.

Com FlexBox você consegue ter uma maleabilidade muito maior no seu layout, sem precisar de hacks, elementos extras ou muitas linhas de CSS. Além disso, como ele é uma tecnologia "nova" _(CSS3)_ é interessante aprender a usá-lo, já que seria uma nova expertise pra você como profissional.

E no caso de precisar suportar navegadores antigos, você ainda tem a possibilidade de usar fallbacks para `float` nos elementos que usam FlexBox.

Se você quiser ver alguns exemplos do _poder_ do FlexBox, recomendo esse link:
https://philipwalton.github.io/solved-by-flexbox/

Para aprender mais sobre as propriedades CSS que o FlexBox engloba, recomendo o link:
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

---

Edit:
Lembrando que essa minha opinião é sobre a melhor forma **hoje** para se desenvolver layout. Isso não significa que não seja importante aprender a usar `float` ou outras formas de montagem de layout mais antigas. Afinal, você pode acabar tendo que fazer manutenção em um site que use essas outras formas e seria interessante entender como tudo ali funciona pra não ter dor de cabeça.

**Comment by thiamsantos** [comment]
@vitocaio você quer entender como funciona o float e o position, ou qual a melhor maneira de fazer layouts? 
Se for a melhor maneira o flexbox tá aí, como disse o @kazzkiq, e no futuro o display grid. Dá uma olhada no #80 para aprender mais de flexbox. 

E  sobre o position tem esse artigo http://tableless.com.br/propriedade-position-do-css/. 
E sobre o float tem esse http://desenvolvimentoparaweb.com/css/css-float-consideracoes-dicas-e-macetes-para-bons-layouts-na-web/

**Comment by vitocaio** [comment]
Só entender qual é melhor mesmo, eu precisei desenvolver um modulo de um sistema e o suporte dele éra IE7+ entao tive q optar em fazer utilizando float ou position, usei float, mas fiquei na duvida se não deveria ter usado position, por conta dos problemas que pode acontecer devido o float ser imprevisível as vezes.

**Comment by iamgbayer** [comment]
A um bom tempo deixei o float de lado e venho utilizando só display, vertical-align e calc(), consigo fazer tranquilamente o que fazia com float, acho que é mais tu ter a criatividade de utilizar essas propriedades que tu consegue substituir numa boa, e caso tu possa estar utilizando o flexbox, utilize.

**Comment by felipefialho** [comment]
Estou utilizando flexbox + calc() com o [Lost](https://github.com/peterramsing/lost) e é uma mão na roda :)

**Comment by rubenmarcus** [comment]
pessoal, float então morreu?
